cudax/include/cuda/experimental/__device/device.cuh

```diff
@@ -48,6 +65,29 @@ public:
   {
     return __id;
   }
+
+  //! @brief Retrieve the specified attribute for the device
+  //!
+  //! @param attr The attribute to query. See `device::attrs` for the available
+  //!        attributes.
+  //!
+  //! @throws cuda_error if the attribute query fails
+  //!
+  //! @sa device::attrs
+  template <::cudaDeviceAttr Attr>
+  _CCCL_NODISCARD auto attr([[maybe_unused]] __attr<Attr> attr) const
```

Comment:
Could we rather use a default argument than a second overload?

Reply:
i could, but i think two overloads is clearer.